cabal.project

```diff
@@ -0,0 +1,2389 @@
+packages:
+-- local packages
+    ./oscoin.cabal
+-- remote packages
+source-repository-package
+  type: git
+  location: https://github.com/oscoin/qc-instances
+  tag: e659eb65a421a91603de4d7645d370abf19aeeef
+source-repository-package
+  type: git
+  location: https://github.com/agrafix/Spock
+  tag: 4eb322398ce9e9b8f0a851955bcc963ae1fedc34
+  subdir: Spock
+source-repository-package
+  type: git
+  location: https://github.com/agrafix/Spock
+  tag: 4eb322398ce9e9b8f0a851955bcc963ae1fedc34
+  subdir: Spock-core
+source-repository-package
+  type: git
+  location: http://github.com/oscoin/radicle
+  tag: eaf70a9e0b36af44d02d0372bc0ac2dcc3d1b29c
+source-repository-package
+  type: git
+  location: https://github.com/oscoin/avl-auth
+  tag: dfc468845a82cdd7d759943b20853999bc026505
+source-repository-package
+  type: git
+  location: https://github.com/oscoin/gossip
+  tag: 78e457dc23f7225d8956381eb5312aa2a0d98356
+source-repository-package
+  type: git
+  location: https://github.com/oscoin/ipfs
+  tag: 30a67159d5b58cf63be903920d6349f0be15cd46
+  subdir: binary-varint
+source-repository-package
+  type: git
+  location: https://github.com/oscoin/ipfs
+  tag: 30a67159d5b58cf63be903920d6349f0be15cd46
+  subdir: ipld-cid
+source-repository-package
+  type: git
+  location: https://github.com/oscoin/ipfs
+  tag: 30a67159d5b58cf63be903920d6349f0be15cd46
+  subdir: multibase
+source-repository-package
+  type: git
+  location: https://github.com/oscoin/ipfs
+  tag: 30a67159d5b58cf63be903920d6349f0be15cd46
+  subdir: multihash-cryptonite
+source-repository-package
+  type: git
+  location: https://github.com/oscoin/ipfs
+  tag: 30a67159d5b58cf63be903920d6349f0be15cd46
+  subdir: multihash-serialise
+
+package sandi
+  flags: -with-conduit 
+
+constraints:
+-- lts-13.10 (except for extra dependencies)
```

Comment:
😱 

Have you tried just placing the `cabal.config` next to this (https://www.stackage.org/lts-13.10/cabal.config)? Would be much easier to maintain...

Reply:
Oh, this bit is done automatically by `stack2cabal` if that is of any relief -- I didn't write any of that!

Reply:
Mhm.. :trollface:

Reply:
And ps, that's essentially what `stack2cabal` does, it [pulls](https://github.com/edsko/stack2cabal/blob/master/src/Stack2Cabal/Resolver.hs#L18-L28) the `cabal.config` from Stackage and dumps it into the `.project` file 😉 It also explains why I had to (provisionally) butcher our original `stack.yaml` -- the tool expects the input `stack.yaml` to contain a `resolver` entry, which, in case of our `stack.yaml`, points to another, nested yaml file (snapshot.yaml), not the actual name of the LTS 😁   

I am open to alternative suggestions, of course, heh.

Reply:
Yeah I'm not sure if it's a good idea to rely on `stack2cabal` if we want to make `cabal` a first-class citizen. Dumping the cabal.config into a separate file (which you can even choose to ignore) seems like a cleaner and more maintainable solution.